### PR TITLE
chore(compile): prefer denort binary in target/ directory when available

### PR DESCRIPTION
**Pull request by irbull**
Enhances the deno compile workflow for denort development by automatically checking for a denort binary in the same directory as the deno binary, provided both are located within a target/ directory. If found, this denort binary will be used.
    
Key points:

- The DENORT_BIN environment variable remains supported for explicitly specifying a custom denort binary path.
- Includes additional logic to  verify if the deno binary is a symlink pointing to an executable in the target/ directory.

### CODE REVIEW

File: cli/standalone/binary.rs [hunk 1]
```diff
@@ -904,6 +906,46 @@ impl<'a> DenoCompileBinaryWriter<'a> {
...
+      } else if exec_path.is_symlink() {
+        fs::read_link(&exec_path).ok().and_then(|target_path| {
+          if is_in_target(&target_path) {
+            try_get_denort(&target_path)
+          } else {
+            None
+          }
+        })
+      } else {
+        None
```
Comment (irbull) [inline cli/standalone/binary.rs:943]: Do you think we should only do this for "debug" builds, or check that the path includes `target/debug`? Or is there value in providing this for release builds too?

File: cli/standalone/binary.rs [hunk 2]
```diff
@@ -904,6 +906,38 @@ impl<'a> DenoCompileBinaryWriter<'a> {
...
+      } else if exec_path.is_symlink() {
+        fs::read_link(&exec_path).ok().and_then(|target_path| {
+          if is_in_target(&target_path) {
+            get_denort(target_path)
+          } else {
+            None
+          }
+        })
+      } else {
+        None
```
Comment (irbull) [inline cli/standalone/binary.rs:935]: Do you think we should only do this for "debug" builds, or check that the path includes target/debug? Or is there value in providing this for release builds too?

File: cli/standalone/binary.rs [hunk 3]
```diff
@@ -904,6 +906,38 @@ impl<'a> DenoCompileBinaryWriter<'a> {
...
+      if is_in_target(&exec_path) {
+        get_denort(exec_path)
+      } else if exec_path.is_symlink() {
+        fs::read_link(&exec_path).ok().and_then(|target_path| {
+          if is_in_target(&target_path) {
+            get_denort(target_path)
+          } else {
+            None
+          }
+        })
```
[the remark this answers is not in the document]
Comment (irbull) [reply cli/standalone/binary.rs]: The symlink check determines if the `deno` executable is in fact _just a sym link_ and if so, it checks that the symbolic link points to an executable in a `target/` directory. This is useful if you link your deno executable from your `cargo/bin` directory to your target directory. 

But you're right, a lot more can go wrong including file handling on different platforms. I've removed this check.

File: cli/standalone/binary.rs (hunk 2 above)
Comment (irbull) [reply cli/standalone/binary.rs]: Let's keep it for `target/` for now and see how it works. We could broaden this in the future to always check for `denort` next to the deno executable. 

Can you test this on Windows? I guess the rt file is called `denort` there too (without a file extension).

File: cli/standalone/binary.rs [hunk 4]
```diff
@@ -904,6 +906,27 @@ impl<'a> DenoCompileBinaryWriter<'a> {
   }
 }
 
+fn get_denort_path(deno_exe: PathBuf) -> Option<OsString> {
+  let mut denort = deno_exe;
+  denort.set_file_name("denort");
```
Comment (dsherret) [inline cli/standalone/binary.rs:911]: > Can you test this on Windows? I guess the rt file is called denort there too (without a file extension).

Good point:

```suggestion
  denort.set_file_name(if cfg!(windows) { "denort.exe" } else { "denort" });
```

Comment (irbull) [reply cli/standalone/binary.rs]: I assumed that `denort` was a binary, non-executable file that was used to link against. I didn't realize it was actually an executable file, now I'm curious A) what it does, and B) how you build the custom compiled artifact. 

Anyways, I've applied this suggestion and fixed the formatting.

**Review by dsherret — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**